### PR TITLE
Issue #443: Rename AbstractAttenuator to AbstractTransmission

### PR DESCRIPTION
**Pull request by MartinSavko**

### CODE REVIEW

**Comment by IvarsKarpics** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- HardwareObjects/abstract/AbstractTransmission.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mxcube/HardwareRepository/pullRequest?prid=4641869&bid=15549563)

**Comment by IvarsKarpics** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- HardwareObjects/abstract/AbstractTransmission.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mxcube/HardwareRepository/pullRequest?prid=4641869&bid=15549563)

**Comment by IvarsKarpics** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- HardwareObjects/abstract/AbstractTransmission.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mxcube/HardwareRepository/pullRequest?prid=4641869&bid=15549563)

**Comment by IvarsKarpics** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- HardwareObjects/abstract/AbstractTransmission.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mxcube/HardwareRepository/pullRequest?prid=4641869&bid=15549563)

**Comment by IvarsKarpics** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- HardwareObjects/abstract/AbstractTransmission.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mxcube/HardwareRepository/pullRequest?prid=4641869&bid=15549563)

File: HardwareObjects/abstract/AbstractTransmission.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+__copyright__ = """ Copyright © 2019 by the MXCuBE collaboration """
+__license__ = "LGPLv3+"
+
+
+class AbstractTransmission(AbstractMotor):
+    """
+    Base class for transmission control by filters, slits, apertures or other means
+    """
+
```
Comment (jordiandreu) [inline HardwareObjects/abstract/AbstractTransmission.py]: We are imposing here that the transmission Hardware Object must implement a motor interface defined in the AbstractMotor class. Could we imagine an scenario in which the transmission is not set by moving a motor? Would then have sense this inheritance?

Comment (IvarsKarpics) [reply HardwareObjects/abstract/AbstractTransmission.py]: I think AbstractTransmission should not be inherited from an AbstractMotor. Base class could be HardwareObject (like AbstractEnergy and AbstractDetector)

Comment (rhfogh) [reply HardwareObjects/abstract/AbstractTransmission.py]: I would say that the base class should be in between AbstractMotor (which has a velocity), and HardwareObject (which has nothing except the config system). Something like AbstractActuator. But @beteva and I will be back on that.

**Comment by marcus-oscarsson** [comment]
This PR was made early in the meeting before we decided to also introduce AbstractActautor at this point in the meeting we **DID** say that Transmission would inherit AbstractMotor since it was the only thing that represented something that moved and had a value. And that the logical gap between HardwareObject and AbstractMotor would need to be filled later or accepted as it was. In this regard the changes are fine and I would not mind merging this and make another issue simply saying replace AbstractMotor with AbstractActuator in AbstractTransmission.

**Comment by marcus-oscarsson** [comment]
@MartinSavko I believe that we are not far from having a PR with AbstractActuator. What would you prefer that we merge this PR as it is or that you update this to replace the Motor inheritance with AbstractActuator ?

**Comment by IvarsKarpics** [comment]
Hi @MartinSavko  and @marcus-oscarsson 

What is the status of this PR. I can merge it and make an another PR that would contain AbstractTrasmission as AbstractActuator

**Comment by marcus-oscarsson** [comment]
@IvarsKarpics That would be a good idea

**Pull request merged**